flink/src/main/java/org/apache/iceberg/flink/data/vectorized/VectorizedFlinkOrcReaders.java

```diff
@@ -0,0 +1,278 @@
+          throw new UnsupportedOperationException(
+              "Unsupported vector: " + vector.getClass().getName());
+        }
+      };
+    }
+  }
+
+  private static class RowPositionColumnVector implements org.apache.flink.table.data.vector.LongColumnVector {
+    private final long batchOffsetInFile;
+
+    RowPositionColumnVector(long batchOffsetInFile) {
+      this.batchOffsetInFile = batchOffsetInFile;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return false;
+    }
+
+    @Override
+    public long getLong(int i) {
+      return batchOffsetInFile + i;
+    }
+  }
+
+  private static class StructConverter implements Converter {
+    private final Types.StructType structType;
+    private final List<Converter> fieldConverters;
+    private final Map<Integer, ?> idToConstant;
+
+    private StructConverter(Types.StructType structType, List<Converter> fieldConverters,
+                            Map<Integer, ?> idToConstant) {
+      this.structType = structType;
+      this.fieldConverters = fieldConverters;
+      this.idToConstant = idToConstant;
+    }
+
+    @Override
+    public ColumnVector convert(org.apache.orc.storage.ql.exec.vector.ColumnVector vector, int batchSize,
+                                long batchOffsetInFile) {
+      StructColumnVector structVector = (StructColumnVector) vector;
+      List<Types.NestedField> fields = structType.fields();
+      ColumnVector[] fieldVectors = new ColumnVector[fields.size()];
+      for (int pos = 0; pos < fields.size(); pos++) {
+        Types.NestedField field = fields.get(pos);
+        if (idToConstant.containsKey(field.fieldId())) {
+          fieldVectors[pos] = toConstantColumnVector(field.type(), idToConstant.get(field.fieldId()));
+        } else if (field.equals(MetadataColumns.ROW_POSITION)) {
+          fieldVectors[pos] = new RowPositionColumnVector(batchOffsetInFile);
+        } else {
+          fieldVectors[pos] = fieldConverters.get(pos)
+              .convert(structVector.fields[pos], batchSize, batchOffsetInFile);
+        }
+      }
+
+      return new FlinkRowColumnVector(fieldVectors, structVector);
+    }
+
+    private ColumnVector toConstantColumnVector(Type type, Object constant) {
+      Type.TypeID typeID = type.typeId();
+      switch (typeID) {
+        case INTEGER:
+        case DATE:
+        case TIME:
+          return ConstantColumnVectors.ints(constant);
+
+        case LONG:
+          return ConstantColumnVectors.longs(constant);
+
+        case BOOLEAN:
+          return ConstantColumnVectors.booleans(constant);
+
+        case DOUBLE:
+          return ConstantColumnVectors.doubles(constant);
+
+        case FLOAT:
+          return ConstantColumnVectors.floats(constant);
+
+        case DECIMAL:
+          return ConstantColumnVectors.decimals(constant);
+
+        case TIMESTAMP:
+          return ConstantColumnVectors.timestamps(constant);
+
+        case FIXED:
+        case UUID:
+        case BINARY:
```

Comment:
I still don't think we could use the `ConstantBytesColumnVector` to read the `BINARY` & `FIXED` data type because its constant object is a [StringData](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java#L66),  we will definitely encounter a class cast exception when cast the `byte[]` to `BinaryStringData` ,  do we have an unit test to cover this ?

Reply:
Be careful that the `FIXED` data type has three constant data type:
* byte[]
* GenerateData.Fixed
* ByteBuffer

We will need to make the `ConstantBytesColumnVector` handle all those types !

Reply:
I added the judgment that constant is byte[]，and add the UT for byte[] data type

Reply:
from the [code](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java#L66) 

> I still don't think we could use the `ConstantBytesColumnVector` to read the `BINARY` & `FIXED` data type because its constant object is a [StringData](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java#L66), we will definitely encounter a class cast exception when cast the `byte[]` to `BinaryStringData` , do we have an unit test to cover this ?

from the [code](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java#L66) ,for `FIXED ` data type,it will be convert to byte[],so I I added the judgment that constant is byte[] instead of throw a cast exception.